### DISCUSSION TITLE
How to adjust the initial thrust value of cf?

### DISCUSSION THREAD

**Discussion by swarm5** [USC-ACTLab/crazyswarm#336] [Q&A]
@whoenig @jpreiss 
I changed the motor with a KV value of 18000, which means the motor has a stronger thrust. When I did the flight test, the flight was relatively stable. But there is a problem: the initial thrust value of cf is too high, so that cf will rush out a certain distance during the take-off stage, then hover, and finally take off smoothly.
**I want to know, which parameter is used to set the initial thrust value of cf in crazyswarm?**

**Comment by whoenig** [comment]
If you are using the mellinger controller, the parameter you are looking for is massThrust (which needs to be lowered for higher KV).

**Reply by swarm5** [reply]
Thank you for your answer! However, if I use a PID controller, how should I adjust the initial thrust value? Because I need to use low-level instructions such as cmdpostion.

**Reply by whoenig** [reply]
I am not sure about the PID controller. You might find help in the official support forum.

**Reply by swarm5** [reply]
I don’t know much about the Mellinger controller, I would like to ask:
The control parameters under the path of `crazyswarm/ros_ws/src/crazyswarm/launch/crazyflieTypes.yaml`, for example:

>          kp_xy: 0.4 #0.6 # p for position
>          kd_xy: 0.2 # 0.35 # d for position
>          ki_xy: 0.05
>          i_range_xy: 2.0
>          kR_xy: 70000 # p for attitude
>          kw_xy: 20000 # d for attitude
>          kR_z: 60000 # 70000 # p for yaw
>          kw_z: 12000 # 30000 # d for yaw
>          ki_m_z: 500 #0 #20000 # i for yaw
>          i_range_m_z: 1500 # i range for yaw
>          kd_omega_rp: 200 # roll and pitch angular velocity d gain
>          kp_z: 1.25
>          kd_z: 0.4
>          ki_z: 0.05
>          i_range_z: 0.4

**Do these parameters affect both PID and Mellinger controllers?
Or, these parameters can only adjust the control performance of the PID controller?**

**Reply by whoenig** [reply]
Just mellinger, because they are in the ctrlMel group, see https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/launch/crazyflieTypes.yaml#L9.
